File: 931. Minimum Falling Path Sum/main.py
class Solution:
    def minFallingPathSum(self, matrix: List[List[int]]) -> int:
        r = len(matrix)
        c = len(matrix[0])
        for i in range(1, r):
            for j in range(c):
                inicio = max(0, j-1)
                fin = min(c-1, j+1)
                matrix[i][j] += min(matrix[i-1][inicio:fin+1])
        
        return min(matrix[-1])

File: 931. Minimum Falling Path Sum/main.js
/**
 * @param {number[][]} matrix
 * @return {number}
 */
var minFallingPathSum = function (matrix) {
  for (let i = 1; i < matrix.length; i++) {
    for (let j = 0; j < matrix[0].length; j++) {
      let min = matrix[i - 1][j];
      if (j - 1 >= 0) {
        min = Math.min(min, matrix[i - 1][j - 1]);
      }
      if (j + 1 < matrix[0].length) {
        min = Math.min(min, matrix[i - 1][j + 1]);
      }
      matrix[i][j] += min;
    }
  }

  return Math.min(...matrix[matrix.length - 1]);
};
